packages/@lwc/synthetic-shadow/src/shared/node-ownership.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty, isNull, isUndefined } from '@lwc/shared';
+import { parentNodeGetter } from '../env/node';
+
+// Used as a back reference to identify the host element
+const HostElementKey = '$$HostElementKey$$';
+const ShadowedNodeKey = '$$ShadowedNodeKey$$';
+
+interface ShadowedNode extends Node {
+    [HostElementKey]: number;
+    [ShadowedNodeKey]: number;
+}
+
+export function getNodeOwnerKey(node: Node): number | undefined {
+    return (node as ShadowedNode)[HostElementKey];
+}
+
+export function setNodeOwnerKey(node: Node, value: number) {
+    const shadowedNode = node as ShadowedNode;
+    if (process.env.NODE_ENV !== 'production') {
+        // in dev-mode, we are more restrictive about what you can do with the owner key
+        defineProperty(shadowedNode, HostElementKey, {
+            value,
+            configurable: true,
+        });
+    } else {
+        // in prod, for better perf, we just let it roll
+        shadowedNode[HostElementKey] = value;
+    }
```

Comment:
Let's extract this block in a `fastDefineProperty` method that can be shared between `setNodeOwnerKey` and `setNodeKey`.